### DISCUSSION TITLE
How to _actually_ ignore files?

### DISCUSSION THREAD

**Discussion by punkpeye** [eslint/eslint#19121] [General]
https://eslint.org/docs/latest/use/configure/ignore

Placing files in `ignores` doesn't entirely ignore the file, e.g. I have a `ignores` rule for `**/*.d.ts`. However, if I run `eslint . --debug`, I see the following in the logs:

```
  eslint:eslint Lint /Users/test/styled-system/types/csstype.d.ts +11ms
  eslint:linter Linting code for /Users/test/styled-system/types/csstype.d.ts (pass 1) +1ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/test/styled-system/types/csstype.d.ts +0ms
  eslint:languages:js Parsing: /Users/test/styled-system/types/csstype.d.ts +9ms
  eslint:languages:js Parsing successful: /Users/test/styled-system/types/csstype.d.ts +94ms
  eslint:languages:js Scope analysis: /Users/test/styled-system/types/csstype.d.ts +0ms
  eslint:languages:js Scope analysis successful: /Users/test/styled-system/types/csstype.d.ts +0ms
[BABEL] Note: The code generator has deoptimised the styling of /Users/test/styled-system/types/csstype.d.ts as it exceeds the max of 500KB.
```

Even though no errors were reported for the file, it still was processed, which is a huge problem for large files.

My configuration:

```js
module.exports = [
  {
    files: ['**/*.cjs', '**/*.ts', '**/*.tsx'],
  },
  // ... rules
  {
    ignores: [
      '**/*.d.ts',
      '**/app/styled-system/',
      '**/build/',
      '**/coverage/',
      '**/database/',
      '**/pnpm-lock.yaml',
      '**/public/',
      '**/server/',
      '**/test-results/',
      '**/vite.config.ts.*',
    ],
    settings: {
      react: {
        version: '19',
      },
    },
  },
];
```

**Comment by fasttime** [comment]
Can you please show your configuration?

**Reply by punkpeye** [reply]
Updated question to include configuration.

**Reply by fasttime** [reply]
Thanks! As indicated in [this section](https://eslint.org/docs/latest/use/configure/ignore#ignoring-files) of the linked documentation page, global ignores patterns are defined by placing an `ignores` key _without any other keys_ in the config object. In your case, you could change your config like this:

```js
module.exports = [
  {
    files: ['**/*.cjs', '**/*.ts', '**/*.tsx'],
  },
  // ... rules
  {
    ignores: [
      '**/*.d.ts',
      '**/app/styled-system/',
      '**/build/',
      '**/coverage/',
      '**/database/',
      '**/pnpm-lock.yaml',
      '**/public/',
      '**/server/',
      '**/test-results/',
      '**/vite.config.ts.*',
    ],
  },
  {
    settings: {
      react: {
        version: '19',
      },
    },
  },
];
```

**Reply by punkpeye** [reply]
Thanks... That said, this is the most unintuitive convention I've ever come across. Like, it could be just named `globalIgnores`. Why this convention?

**Reply by fasttime** [reply]
> Thanks... That said, this is the most unintuitive convention I've ever come across. Like, it could be just named `globalIgnores`. Why this convention?

I believe in the [original design](https://github.com/eslint/rfcs/blob/main/designs/2019-config-simplification/README.md?plain=1#L538) `ignores` pattern would act as global ignores whenever a config specified no `files`, but that was changed later to permit certain use cases.
